### ISSUE TITLE
Upgrade to use USGS Spectral Library Version 7  

### ISSUE THREAD

**Issue by lewismc** [capstone-coal/pycoal#105]
Spectral Lib 7 is now available
https://speclab.cr.usgs.gov/spectral-lib.html

**Comment by lewismc** [comment]
I can't seem to find the equivalent version of ftp://ftpext.cr.usgs.gov/pub/cr/co/denver/speclab/pub/spectral.library/splib06.library/Convolved.libraries/s06av95a_envi.hdr

**Comment by ghost** [comment]
It [looks like](https://www.sciencebase.gov/catalog/item/5807a2a2e4b0841e59e3a18d) they have not released an AVIRIS/ENVI version of the library, but that's okay, we'll just have to use Spectral to convert it by doing something similar or identical to [`pycoal.mineral.AsterConversion`](https://pycoal.readthedocs.io/en/latest/mineral.html#pycoal.mineral.AsterConversion).

Rather than making COAL users do it, we might consider converting it ourselves and then hosting it since it is (I presume) freely distributable.

**Comment by lewismc** [comment]
There will be a requirement to update the Website documentation here as well.

**Comment by lewismc** [comment]
Hi @bdegley4789 how are things coming along? Anything I can help out with over and above out meeting last week? Do you have a pull request I can review? Thanks

**Comment by bdegley4789** [comment]
@lewismc the last few days, I was working on fixing the errors I was having with gdal, which I solved. And I think I figured out what was causing the rotational bug with qgis. I added these to the installation instructions and made a pull request on the website for that. On this issue, I've looked at the links you sent but not much other than that. This will be my new priority now that the gdal stuff is out of the way. I'll get something ready for this issue either in the next day or two at least before our next meeting

**Comment by bdegley4789** [comment]
@lewismc Do you know where the Aster data files are located in usgs spectral library version 7? I keep getting an error that they aren't found when I set the path to several different locations in usgs_splib07. I set up this code to use pycoal.mineral.AsterConversion and generates s07av95a_envi.hdr and s07av95a_envi.sli  
![1](https://user-images.githubusercontent.com/25111791/37163874-bff8d648-22ad-11e8-93d7-6ef0486a28b6.png)

**Comment by bdegley4789** [comment]
Current code
`#!/usr/bin/python
#
# Copyright (C) 2018 COAL Developers
#
# This program is free software; you can redistribute it and/or 
# modify it under the terms of the GNU General Public License 
# as published by the Free Software Foundation; version 2.
#
# This program is distributed in the hope that it will be useful, 
# but WITHOUT ANY WARRANTY; without even the implied warranty 
# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
# See the GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public 
# License along with this program; if not, write to the Free 
# Software Foundation, Inc., 51 Franklin Street, Fifth 
# Floor, Boston, MA 02110-1301, USA.
# encoding: utf-8

'''
example_sam -- Plot Spectral Angles

.sli and .hdr files will be generated from spectral library.

@author:     COAL Developers

@copyright:  2018 COAL Developers. All rights reserved.

@license:    GNU General Public License version 2

@contact:    coal-capstone@googlegroups.com
'''


from argparse import ArgumentParser
from argparse import RawDescriptionHelpFormatter

import logging

import sys
import os
from sys import path
from os import getcwd
import inspect
import pycoal
sys.path.insert(0, '../pycoal')
import mineral
import math
import numpy
import spectral

# load library
library_filename = '../pycoal/tests/usgs_splib07/ASCIIdata/ASCIIdata_splib07b_cvASD/ChapterA_ArtificialMaterials'
data_dir = "dataSqlite07"
header_name = "s07av95a_envi"

# create a new mineral aster conversion instance
spectral_envi = mineral.AsterConversion()
# Generate .sli and .hdr
spectral_envi.convert(library_filename,data_dir,header_name)`

**Comment by bdegley4789** [comment]
Current code 
https://github.com/bdegley4789/pycoal/blob/master/examples/example_splib07.py

**Comment by bdegley4789** [comment]
If pycoal.mineral.AsterConversion doesn't work I think we could also use [Harris Geospatial Spectral Library Builder](https://www.harrisgeospatial.com/docs/spectrallibraries.html)

**Comment by ghost** [comment]
@bdegley4789 Since @lewismc hasn't responded yet, I recommend you have a look at the [source code of `pycoal.mineral.AsterConversion`](https://github.com/capstone-coal/pycoal/blob/271c0d3031d98a711a57bb4087b75f7b2f0aed5f/pycoal/mineral.py#L316-L364). This is an experimental method expected to work only on the ASTER spectral library. The USGS Spectral Library Version 7 uses a similar but distinct format. You may also need to review `spectral.AsterDatabase.create_envi_spectral_library()` to see how to go from an array of samples to a resampled library. The proprietary ENVI software from Harris Geospatial would also certainly work, but if you have a look at the licensing you will see why we haven't used it.

**Comment by bdegley4789** [comment]
@browtayl Thanks. Lewis didn't responded here but we discussed this in the video chat meeting we had last Thursday. He had similar recommendations, I'll take a look at `spectral.AsterDatabase.create_envi_spectral_library()` I didn't know about that

**Comment by lewismc** [comment]
How are things coming along @bdegley4789 ?

**Comment by lewismc** [comment]
PING @bdegley4789

**Comment by bdegley4789** [comment]
@lewismc sorry for the late reply I had a project in another class yesterday. I am working on this by creating another convert class like we discussed. With the AsterConversion it used .db file. The usgs Spectral Version 7 is a bunch of .txt files I'm a little confused on how to work with this difference. I think I can still work around that to create the .sli and .hdr files but Im not sure if the contents with in them will be whats needed

**Comment by lewismc** [comment]
@bdegley4789 thanks, the information you need to get your started can be found at http://www.spectralpython.net/libraries.html. This will lead you to working with the [SpectralLibrary Class](http://www.spectralpython.net/class_func_ref.html#spectral.io.envi.SpectralLibrary). The output of the save function in this Class is two files: fileBaseName.hdr and fileBaseName.sli.
Any issues, let me know and we can work it through.

**Comment by lewismc** [comment]
@bdegley4789 @thomkenn please keep me updated this next week. I look forward to us making progress as a team.

**Comment by bdegley4789** [comment]
Sounds like a plan, Ill keep you updated

**Comment by bdegley4789** [comment]
Ok I worked on this issue for awhile last night and today. I found a way to get the .sli and .hdr files of all the Spectra in Spectral Library Version 7.

If I can change all the files in Spectral Library Version 7 to the format of the ASTER files and move them to one directory and then run
the AsterConversion() function in mineral.py in my most recent pull request to pycoal it does correctly output the .hdr and .sli files. 

I have tested this with one Spectral Library 7 .txt file which I manually edited to have the same format as the ASTER text files. You do this by running example_splib07.py in the example directory and it will output a database file, .sli and .hdr for the one Spectral 7 file I formatted to match ASTER file format. And it appears correctly in the .hdr file

There is a slight problem I think we can work around by doing this method
Aster Files require:
Name
Particle Size
Sample No.
Wavelength
Origin
Measurement
Two Columns and the Units of Measurement

-Spectral Version 7 is all in micrometers same as ASTER
-The name will be the name of the Spectral Version 7 mineral
-On the other values though Spectral Version 7 doesn’t give that information. 
However, in mineral.py we are only sorting the x column so whatever value I put in for the y column and those other values should be irrelevant. 
If this works for you @lewismc , I’ll move forward and write the code to convert all the Spectral Text files to match Aster file format and then get the .hdr and .sli files for the entire spectral version 7 library

I believe this method can also be applied to solving issues #134  and #119 since those spectra are .txt files of a different format.

**Comment by bdegley4789** [comment]
Ok, I went ahead with the method described above. 
I have the envi .hdr and .sli files generated for the entire USGS Spectral Library Version 7. 
I have also started to run them in a mineral classification of a site to see how it will look using Spectral Version 7. I pushed all the code I wrote to do this in my most recent pull request and staged the .hdr and .sli files in https://drive.google.com/drive/folders/1YVhdLxvrZE3eC97OEXathLMJRgWt8haO

This next week I will create two more convert functions to work on the EcoSIS and ECOSTRESS spectral libraries
